CowLite Audioplayer/src/cap/gui/shared/Slider.java

```diff
@@ -0,0 +1,149 @@
+package cap.gui.shared;
+
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseListener;
+import java.awt.event.MouseMotionListener;
+import javax.swing.JComponent;
+import javax.swing.event.ChangeEvent;
+import javax.swing.event.ChangeListener;
+
+/**
+ * (c) Copyright To make a better looking timebar
+ */
+public class Slider extends JComponent implements MouseMotionListener, MouseListener {
+    
+    // MARK: - Associated types & constants
+    
+    public enum Orientation {
+        horizontal,
+        vertical
+    }
+
+    private double value = 0;
+    private double minimumValue = 0;
+    private double maximumValue = 0;
+    private ChangeListener listener;
```

Comment:
Should be weakified. Also, consider replacing `ChangeListener` with some custom delegate implementation.